haskell/src/Monocle/Class.hs

```diff
@@ -101,7 +101,7 @@ instance MonadLog IO where
 -------------------------------------------------------------------------------
 -- A http system
 
-class (MonadRetry m, MonadLog m) => MonadGraphQL m where
+class (MonadRetry m, MonadLog m, MonadUnliftIO m) => MonadGraphQL m where
```

Comment:
Ideally we shouldn't enable IO, but we would need to define a class for `newMVar` and `modifyMVar`. But perhaps it is not so useful to have those classes and we could use a simpler `MonadUnlinftIO` constraint.

Thus I suggest we comment here that this is needed for Concurrent.MVar, and then in a follow-up we can clean-up this module.

Reply:
Yes actually I started to do that in my first implementation. So I bring it back. I think that better now.